Sources/Observation.swift

```diff
@@ -95,6 +99,30 @@ extension Defaults {
 		public func invalidate() {
 			object?.removeObserver(self, forKeyPath: key, context: nil)
 			object = nil
+			lifetimeTie = nil
+		}
+
+		private struct LifetimeTie {
+			weak var object: AnyObject?
+		}
+
+		private var lifetimeTie: LifetimeTie?
+		private static var lifetimeTieAssociationKey = AssociatedObject<UserDefaultsKeyObservation>()
+
+		public func tieToLifetimeOf(_ weaklyHeldObject: AnyObject) -> DefaultsObservation {
```

Comment:
What do you think of implementation these two methods as default implementations of the protocol instead?